### PR TITLE
Adds calendar view to schedule page

### LINKED ISSUE #1215: Schedule: Implement a calendar view
Currently a proper calendar view of the schedule is missing. Please implement a view similar to the calendar view of the web app.

On each page give the option 
* List View
* Calendar View

### PR DESCRIPTION
**Pull request by geekyd**
Fixes #1215

Changes: Implements a new view in schedule page for viewing the sessions. The user can now switch between the list and calendar(Grid) view using the toggle buttons. 

Demo server :
https://open-event-web-appp.herokuapp.com

@mariobehling @aayusharora @Princu7 Please review this. Thanks :)
Screenshots for the change: 
screen shot 2017-04-24 at 1 06 02 am
screen shot 2017-04-24 at 1 06 39 am

### CODE REVIEW

File: src/backend/fold.js [hunk 1]
```diff
@@ -42,6 +43,54 @@ function checkNullHtml(html) {
...
+function getTimeDifferenceOfDates(startTime, sessionTime) {
+  const firstDate = new Date(new Date().getFullYear(), new Date().getMonth(), new Date().getDate(), parseInt(getHoursFromTime(startTime)), parseInt(getMinutessFromTime(startTime)), 0);
+  const secondDate = new Date(new Date().getFullYear(), new Date().getMonth(), new Date().getDate(), parseInt(getHoursFromTime(sessionTime)), parseInt(getMinutessFromTime(sessionTime)), 0);
+  return (secondDate - firstDate);
+}
+
+function convertTimeToPixel(startTime, sessionTime) {
+  const timeDiff = getTimeDifferenceOfDates(startTime, sessionTime);
+
+  const top = timeDiff * timeToPixel/ (1000 * 60 * 15) + timeToPixel; // distance of session from top of the table
```
Comment (aayusharora) [inline src/backend/fold.js]: @geekyd How have you calculated the top? According to my approach =>
1. We should see the box height in pixels, let's say it as 100 px in height.
2. Now each 15minutes ==> 100 px height. 1 min  = 100/15 px, let say it s.
3. Calculating date difference, changing into minutes and multiplying by s.
 @geekyd Is your approach different ?

Comment (geekyd) [inline src/backend/fold.js]: @aayusharora That is exactly what I'm doing.

1. Get the start time of the first event and use it to calculate top for each session.
2. We are using `timeToPixel ` which is used to keep the pixel value corresponding to 15 mins.
3. We take time difference in `timeDiff` relative to the start time we saved earlier, and compute the top using formula `timeDiff * timeToPixel / (1000 * 60 * 15) + timeToPixel;`

Explanation : 

- `timeDiff` is difference in time in ms, we must convert it to minutes, we divide `timeDiff` by 1000 * 60 to convert it to minutes

- we add `timeToPixel` to the resulting value to make sure the the sessions start after the room heading. All room headings have `height = timeToPixel`

Comment (aayusharora) [inline src/backend/fold.js]: Sounds good @geekyd Thanks

**Comment by aayusharora** [comment]
LGTM :)

**Comment by mariobehling** [comment]
This is a great feature, but it is too buggy and we are already using this in production. Please help to fix all issues of the calendar view.

**Comment by geekyd** [comment]
@mariobehling Sorry for the problems caused I'm working on it.

**Pull request merged**